ark/utils/load_utils.py

```diff
@@ -297,37 +315,43 @@ def load_imgs_from_dir(data_dir, imgdim_name='compartments', image_name='img_dat
                           f"because the loaded images are floats")
             dtype = data_dtype
 
-    if variable_sizes:
-        img_data = np.zeros((len(imgs), 1024, 1024, 1), dtype=dtype)
-    else:
-        img_data = np.zeros((len(imgs), test_img.shape[0], test_img.shape[1], 1),
-                            dtype=dtype)
+    # extract data
+    img_data = []
+    for img in imgs:
+        v = io.imread(os.path.join(data_dir, img))
+        if v.ndim == 2:
+            v = v[:, :, np.newaxis]
+        elif v.shape[0] == min(v.shape):
+            # covert channels_first to be channels_last
+            v = np.moveaxis(v, 0, -1)
+        img_data.append(v)
+    img_data = np.stack(img_data, axis=0)
+
+    img_data = img_data.astype(dtype)
 
-    for img in range(len(imgs)):
-        if variable_sizes:
-            temp_img = io.imread(os.path.join(data_dir, imgs[img]))
-            img_data[img, :temp_img.shape[0], :temp_img.shape[1], 0] = temp_img.astype(dtype)
-        else:
-            img_data[img, :, :, 0] = io.imread(os.path.join(data_dir, imgs[img])).astype(dtype)
+    if channel_indices:
+        img_data = img_data[:, :, :, channel_indices]
 
     # check to make sure that dtype wasn't too small for range of data
     if np.min(img_data) < 0:
         raise ValueError("Integer overflow from loading TIF image, try a larger dtype")
 
-    if variable_sizes:
-        row_coords, col_coords = range(1024), range(1024)
+    if test_img.ndim == 3 and test_img.shape[0] == min(test_img.shape):
+        # channels_first
+        row_coords, col_coords = range(test_img.shape[1]), range(test_img.shape[2])
     else:
         row_coords, col_coords = range(test_img.shape[0]), range(test_img.shape[1])
 
     # get fov name from imgs
     fovs = iou.extract_delimited_names(imgs, delimiter=delimiter)
 
-    img_xr = xr.DataArray(img_data.astype(dtype),
-                          coords=[fovs, row_coords, col_coords, [image_name]],
-                          dims=["fovs", "rows", "cols",
-                          imgdim_name])
+    # create xarray with image data
+    img_xr = xr.DataArray(img_data,
```

Comment:
We should check that channel_names has the same length as the number of channels in the image, and raise an error if not

Reply:
added